src/main/java/com/coreos/jetcd/LeaseImpl.java

```diff
@@ -216,16 +216,31 @@ public synchronized void cancelKeepAlive(long leaseId)
    */
   @Override
   public CompletableFuture<LeaseKeepAliveResponse> keepAliveOnce(long leaseId) {
+    CompletableFuture<LeaseKeepAliveResponse> lkaFuture = new CompletableFuture<>();
 
-    /**
-     * TODO: finish this.
-     */
     StreamObserver<LeaseKeepAliveRequest> requestObserver = this.leaseStub
-        .leaseKeepAlive(keepAliveResponseStreamObserver);
+        .leaseKeepAlive(new StreamObserver<LeaseKeepAliveResponse>() {
+          @Override
+          public void onNext(LeaseKeepAliveResponse leaseKeepAliveResponse) {
+            lkaFuture.complete(leaseKeepAliveResponse);
+          }
+
+          @Override
+          public void onError(Throwable throwable) {
+            lkaFuture.completeExceptionally(throwable);
+          }
+
+          @Override
+          public void onCompleted() {
+          }
+        });
     requestObserver.onNext(newKeepAliveRequest(leaseId));
-    requestObserver.onCompleted();
 
-    throw new UnsupportedOperationException();
+    // cancel grpc stream when leaseKeepAliveResponseCompletableFuture completes.
+    // TODO: manage Async threads.
```

Comment:
I think we should have an internal executor to handle the async threads. So when we close the client, we know where all our threads all. By default, the whenCompleteAsync without a exectuor use a common thread pool. 

"All async methods without an explicit Executor argument are performed using the ForkJoinPool.commonPool()" 
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html